platform/api-server/api/controllers/platform-api-controller.js

```diff
@@ -99,3 +103,244 @@ exports.updatePrivacyPolicy = async function(req, res) {
+        (gwa, gwb) => gwb.resources.memoryFreeMB - gwa.resources.memoryFreeMB,
+        (gwa, gwb) => {
+            const reduceGpuMem = (acc, gpu) => acc + gpu.memoryFreeMB;
+            const a = gwa.resources.gpus.reduce(reduceGpuMem, 0);
+            const b = gwb.resources.gpus.reduce(reduceGpuMem, 0);
+            return b - a;
+        },
+        (gwa, gwb) => gwa.resources.storageFreeMB > gwb.resources.storageFreeMB,
+        (gwa, gwb) => {
+            const reduceGpuUtil = (acc, gpu) => acc + gpu.utilization;
+            const a = gwa.resources.gpus.reduce(reduceGpuUtil, 0);
+            const b = gwa.resources.gpus.reduce(reduceGpuUtil, 0);
+            return b - a;
+        },
+        (gwa, gwb) => gwb.resources.cpuFreePercent - gwa.resources.cpuFreePercent,
+        // Prefer gateways with more of a type of connected devices requested of the application.
+        (gwa, gwb) => {
+            // Accumulate the counts of all the device types that the application will make use of.
+            const sumDeviceTypes = function (gw) {
+                return Object.keys(gw.deviceTypes)
+                    .filter((deviceType) => {
+                        return deviceType in deployMetadata.devices.types
+                            || deviceType in runMetadata.devices;
+                    })
+                    .reduce((count, deviceType) => count + gw.deviceTypes[deviceType], 0);
+            };
+
+            const gwaSum = sumDeviceTypes(gwa);
+            const gwbSum = sumDeviceTypes(gwb);
+
+            return gwbSum - gwaSum;
+        },
+        // Prefer gateways with the most specifically requested devices connected.
+        (gwa, gwb) => {
+            const sumRequestedDevices = function (gw) {
+                    return gw.deviceIds.reduce(
+                        (acc, id) => {
+                            if (requestedDeviceIds.has(id)) {
+                                return acc + 1;
+                            } else {
+                                return acc;
+                            }
+                        }, 0);
+            };
+
+            const gwaCount = sumRequestedDevices(gwa);
+            const gwbCount = sumRequestedDevices(gwb);
+
+            return gwbCount - gwaCount;
+        },
+        // Look for a tighter fit on gateway requirements by prioritizing gateways with the least no. of capabilities.
+        (gwa, gwb) => capability_count(gwa.resources) < capability_count(gwb.resources)
+    ];
+    optimizationSortFns.forEach((sortFn) => { candidates.sort(sortFn); });
+
+    if (candidates.length > 0) {
+        return candidates[0];
+    } else {
+        return null;
+    }
+}
+
+function evaluate_capability(gw, tag) {
+    if (r == 'gpu') {
+        // At least one GPU must have sufficient memory and idle compute.
+        return gw.gpus.reduce((acc, cur) => {
+            return acc || (cur.memoryFreeMB > 200 && cur.utilization < 80);
+        }, false);
```

Comment:
can be simplified as:
return gw.gpus.some(gpu => gpu.memoryFreeMB > 200 && cur.utilization < 80)